### PR TITLE
Remove last focused element on drawer close; focus drawer if tabindex is set

### PR DESCRIPTION
**Pull request by keanulee**
cc @frankiefu @blasten 

You can test the behavior in app-drawer/demo/demo2.html by adding `tabindex="0"` to `<app-drawer>`.

### CODE REVIEW

File: app-drawer/app-drawer.html [hunk 1]
```diff
@@ -552,12 +550,11 @@
 
         if (oldState !== this._drawerState) {
           if (this._drawerState === this._DRAWER_STATE.OPENED) {
+            this._setKeyboardFocusTrap();
             document.addEventListener('keydown', this._boundEscKeydownHandler);
```
Comment (blasten) [inline app-drawer/app-drawer.html]: The current code adds two `keydown` events listeners. I think all the work could be done on a single event listener. Also, there are a few events listeners that are added in the `attached` callback that aren't removed in the `detached` callback.

File: app-drawer/app-drawer.html [hunk 2]
```diff
@@ -593,17 +588,20 @@
...
         }
-      },
 
-      _removeKeyboardFocusTrap: function() {
-        if (!this.noFocusTrap && this._lastFocusedElement) {
-          this._lastFocusedElement.focus();
+        // Focus on app-drawer if it has non-zero tabindex. Otherwise, focus the first focusable
+        // element in the drawer, if it exists. Use the tabindex attribute since the this.tabIndex
+        // property in IE/Edge returns 0 (instead of -1) when the attribute is not set.
+        var tabindex = this.getAttribute('tabindex');
```
Comment (blasten) [inline app-drawer/app-drawer.html]: `this.tabIndex` is already a `Number`

Comment (blasten) [inline app-drawer/app-drawer.html]: q: if I set the attribute then I would get the right `tabIndex` via property in IE/Edge?

File: app-drawer/app-drawer.html (hunk 1 above)
Comment (keanulee) [inline app-drawer/app-drawer.html]: The 2 listeners are installed on different targets (document and this). The document listener is removed in detached, and the element listener will be GC'd

File: app-drawer/app-drawer.html (hunk 2 above)
Comment (keanulee) [inline app-drawer/app-drawer.html]: In IE/Edge, no tabindex attribute and `tabindex="0"` both make the `tabIndex` property `0`, so there's no way of distinguishing between the two (which we need).

**Comment by frankiefu** [comment]
LGTM

**Comment by keanulee** [comment]
@frankiefu Let's tag a new release so we can use this in other projects.

**Comment by frankiefu** [comment]
done

**Pull request merged**